File: LeetCode/Mock Interviews/arrays/median_of_two_sorted_arrays_hari.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int s1=nums1.size();
        int s2=nums2.size();
        int s3;
        bool flag;
        if((s1+s2)%2 !=0)
        {
            s3=((s1+s2)/2);
            flag=true;
        }
        else
        {
            s3=(int)((s1+s2)/2)+1;  // Concern
            flag=false;
        }
        //cout<<s3<<" "<<flag<<endl;
        //return s3;
        int i=0;
        int j=0;
        int k=0;
        vector<int>ans;
        while(i<=s3 && j<s1 && k<s2)
        {
            if(nums1[j]<nums2[k])
            {
                ans.push_back(nums1[j]);
                j++;
            }
            else
            {
                ans.push_back(nums2[k]);
                k++;
            }
            i++;
        }
        while(i<=s3 && j<s1)
        {
            ans.push_back(nums1[j]);
            j++;
            i++;
        }
        while(i<=s3 && k<s2)
        {
            ans.push_back(nums2[k]);
            k++;
            i++;
        }      
        // for(i=0;i<ans.size();i++)
        // {
        //     cout<<ans[i]<<" ";
        // }
        //cout<<endl;
        if(flag)
        {
            return ans[s3];
        }
        else
        {
            //cout<<ans[s3-1]<<" "<<ans[s3-2]<<" "<<((ans[s3-2]+ans[s3-1])/2);
            return((float)(ans[s3-2]+ans[s3-1])/2);
        }
        
    }
};

File: LeetCode/Mock Interviews/arrays/median_of_two_sorted_arrays_hari.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:    
#         def merge(left, right):
#             a = []
#             i,j=0,0
            
#             nl = len(left)
#             nr = len(right)
            
#             while i<nl and j<nr:
#                 if left[i] <= right[j]:
#                     a.append(left[i])
#                     i+=1
#                 else:
#                     a.append(right[j])
#                     j+=1
                    
#             while i<nl:
#                 a.append(left[i])
#                 i+=1
#             while j<nr:
#                 a.append(right[j])
#                 j+=1
                
#             return a
        
#         lst = merge(nums1, nums2)
        
#         l = len(lst)
#         mid = int(l/2)
        
#         if l%2==0:
#             return((lst[mid-1]+lst[mid])/2)
#         else:
#             return(lst[mid])
        
#         return 0.0     

        s1=len(nums1)
        s2=len(nums2)
        if (s1+s2)%2 !=0:
        
            s3=int(((s1+s2)/2))
            flag=True
        else:
            s3=int(((s1+s2)/2))+1  
            flag=False
        i=0
        j=0
        k=0
        ans=[]
        while(i<=s3 and j<s1 and k<s2):
            if(nums1[j]<nums2[k]):
                ans.append(nums1[j])
                j+=1
            else:
                ans.append(nums2[k])
                k+=1
            i+=1
        while(i<=s3 and j<s1):
            ans.append(nums1[j])
            j+=1
            i+=1
        while(i<=s3 and k<s2):
            ans.append(nums2[k])
            k+=1
            i+=1      
        if(flag):
            return ans[s3]
        else:
            return(float((ans[s3-2]+ans[s3-1]))/2)
        
        
        
